### PR TITLE
Generic pointer procs now error if no types supplied

### LINKED ISSUE #16124: Proc assignments with union type classes cause internal errors
### Example
```
proc foo(x: int | float): float = result = 1.0
let bar = foo
```

gives 

```
.......
Error: internal error: getTypeDescAux(tyOr)
No stack traceback available
To create a stacktrace, rerun compilation with './koch temp c <file>', see https://nim-lang.github.io/Nim/intern.html#debugging-the-compiler for detail
```

And a similar example,

```
proc foo(x: int | float): float | int = result = 1.0
let bar = foo
```

Gives 

```
.......fatal.nim(49)            sysFatal
Error: unhandled exception: ccgtypes.nim(200, 17) `false` mapType [AssertionDefect]
```

### Expected Output

I _believe_ these examples should compile, given the expected results without the union types 

```
proc foo(x: int): float = result = 1.0
let bar = foo
echo bar(5)
echo foo(5)
```
does.

### Additional Information

```
λ nim -v
Nim Compiler Version 1.5.1 [MacOSX: amd64]
Compiled at 2020-11-25
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 19e224866ba6cfe2c6cdc49e435a275b532f0acb
active boot switches: -d:release
```

v1.4.0 has the same errors

### PR DESCRIPTION
**Pull request by beef331**
Closes: https://github.com/nim-lang/Nim/issues/16124
(Dont know if this is the smartest solution)

### CODE REVIEW

File: tests/generics/tpointerprocs.nim [hunk 1]
```diff
@@ -0,0 +1,28 @@
+discard """
+cmd: "nim check $options --hints:off $file"
+action: "reject"
+nimout:'''
+tpointerprocs.nim(15, 11) Error: 'foo' doesn't have a concrete type, due to unspecified generic parameters.
+tpointerprocs.nim(27, 11) Error: cannot instantiate: 'foo[int]'; got 1 typeof(s) but expected 2
+tpointerprocs.nim(27, 14) Error: expression 'foo[int][int]' has no type (or is ambiguous)
```
Comment (Araq) [inline tests/generics/tpointerprocs.nim]: Why `foo[int][int]`?

Comment (beef331) [reply tests/generics/tpointerprocs.nim]: It's nim check giving out all the errors, so `foo[int]` got turned into `foo[int][int]`, should I change the error max to the number of expected errors?

Comment (Araq) [reply tests/generics/tpointerprocs.nim]: Well yes, I know but why got it turned into this?

Comment (beef331) [reply tests/generics/tpointerprocs.nim]: So it looks like this is due to https://github.com/nim-lang/Nim/blob/30d28bcefcad0da8900cfa231be9d77bb98c5097/compiler/semexprs.nim#L1513-L1516
What's done in the if statement adds the `[int]` on it again. Seems atleast with a minimal example the following resolves it:
```nim
if result == n: n[0] = copyTree(result[0])
else: n[0] = result 
```

**Pull request merged**